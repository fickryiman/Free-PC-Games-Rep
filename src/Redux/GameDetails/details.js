import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';

const options = {
  method: 'GET',
  headers: {
    'X-RapidAPI-Key': '8599077a4emsh479efd2103792b1p1a37d1jsn1b71b1b00798',
    'X-RapidAPI-Host': 'free-to-play-games-database.p.rapidapi.com',
  },
};

const url = 'https://free-to-play-games-database.p.rapidapi.com/api/game?id';

export const fetchGameDetails = createAsyncThunk('fetch-details', async (id) => {
  const response = await fetch(`${url}=${id}`, options);
  const result = await response.json();
  return result;
});

const gameDetailSlice = createSlice({
  name: 'gameDetails',
  initialState: {
    gameDetails: [],
    loading: true,
  },
  extraReducers: (builder) => {
    builder.addCase(fetchGameDetails.fulfilled, (state, action) => ({
      ...state,
      gameDetails: action.payload,
      loading: false,
    }));
  },
});

export default gameDetailSlice.reducer;
