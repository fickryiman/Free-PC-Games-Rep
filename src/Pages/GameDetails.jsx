import { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { useParams } from 'react-router-dom';
import DOMPurify from 'dompurify';

import LoadingSpinner from '../Components/Loader';
import { fetchGameDetails } from '../Redux/GameDetails/details';

const Details = () => {
  const dispatch = useDispatch();
  const { gameId } = useParams();
  const { gameDetails, loading } = useSelector((state) => state.details);

  useEffect(() => {
    dispatch(fetchGameDetails(gameId));
  }, []);

  return (
    <div className="details-container limit">
      {loading ? (
        <LoadingSpinner />
      ) : (
        <div className="header-game">
          {/* <h1 className="details-header">{` ${gameDetails.title.toUpperCase()}`}</h1> */}
          <img
            className="details-image"
            src={gameDetails.thumbnail}
            alt="games thumbnail"
          />
          <div className="screenshots">
            <img
              className="screenshots-image"
              src={gameDetails.screenshots[0].image}
              alt="game screenshot 1"
            />
            <img
              className="screenshots-image"
              src={gameDetails.screenshots[1].image}
              alt="game screenshot 2"
            />
            <img
              className="screenshots-image"
              src={gameDetails.screenshots[2].image}
              alt="game screenshot 3"
            />
          </div>
          <h2
            className="details-description"
            dangerouslySetInnerHTML={{
              __html: DOMPurify.sanitize(gameDetails.short_description),
            }}
          />
          <div className="details-container">
            <ul className="details-list">
              <li className="details-list-item">
                <h3>Game Name :</h3>
                <h2>{`${gameDetails.title}`}</h2>
              </li>
              <li className="details-list-item">
                <h3>Game Genre :</h3>
                <h2>{`${gameDetails.genre}`}</h2>
              </li>
              <li className="details-list-item">
                <h3>Game Publisher :</h3>
                <h2>{`${gameDetails.publisher}`}</h2>
              </li>
              <li className="details-list-item">
                <h3>Game Developer :</h3>
                <h2>{`${gameDetails.developer}`}</h2>
              </li>
              <li className="details-list-item">
                <h3>Game Release Date :</h3>
                <h2>{`${gameDetails.release_date}`}</h2>
              </li>
              <li className="details-list-item">
                <h3>Game Official Website :</h3>
                <h2>{`${gameDetails.game_url}`}</h2>
              </li>
            </ul>
            <br />
            <br />
            <br />
            <br />
          </div>
        </div>
      )}
    </div>
  );
};

export default Details;
