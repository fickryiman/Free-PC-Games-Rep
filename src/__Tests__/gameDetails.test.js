import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../Redux/storages';
import GameDetails from '../Pages/GameDetails';

describe('Tests for the game details page', () => {
  it('Should render the detail page properly', () => {
    const pageView = render(
      <Provider store={store}>
        <Router>
          <GameDetails />
        </Router>
      </Provider>,
    );
    expect(pageView).toMatchSnapshot();
  });
});
