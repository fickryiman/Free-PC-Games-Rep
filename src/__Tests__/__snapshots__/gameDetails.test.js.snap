// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for the game details page Should render the detail page properly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="details-container limit"
      >
        <div
          class="loader flex"
        >
          <div
            aria-busy="true"
            aria-label="three-circles-rotating"
            class=""
            data-testid="three-circles-wrapper"
            role="status"
            style="display: flex;"
          >
            <svg
              data-testid="three-circles-svg"
              enable-background="new 0 0 100 100"
              height="100"
              version="1.1"
              viewBox="0 0 100 100"
              width="100"
              xml:space="preserve"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M31.6,3.5C5.9,13.6-6.6,42.7,3.5,68.4c10.1,25.7,39.2,38.3,64.9,28.1l-3.1-7.9c-21.3,8.4-45.4-2-53.8-23.3
  c-8.4-21.3,2-45.4,23.3-53.8L31.6,3.5z"
                fill="white"
              >
                <animatetransform
                  attributeName="transform"
                  attributeType="XML"
                  dur="2s"
                  from="0 50 50"
                  repeatCount="indefinite"
                  to="360 50 50"
                  type="rotate"
                />
              </path>
              <path
                d="M42.3,39.6c5.7-4.3,13.9-3.1,18.1,2.7c4.3,5.7,3.1,13.9-2.7,18.1l4.1,5.5c8.8-6.5,10.6-19,4.1-27.7
  c-6.5-8.8-19-10.6-27.7-4.1L42.3,39.6z"
                fill="white"
              >
                <animatetransform
                  attributeName="transform"
                  attributeType="XML"
                  dur="1s"
                  from="0 50 50"
                  repeatCount="indefinite"
                  to="-360 50 50"
                  type="rotate"
                />
              </path>
              <path
                d="M82,35.7C74.1,18,53.4,10.1,35.7,18S10.1,46.6,18,64.3l7.6-3.4c-6-13.5,0-29.3,13.5-35.3s29.3,0,35.3,13.5
  L82,35.7z"
                fill="white"
              >
                <animatetransform
                  attributeName="transform"
                  attributeType="XML"
                  dur="2s"
                  from="0 50 50"
                  repeatCount="indefinite"
                  to="360 50 50"
                  type="rotate"
                />
              </path>
            </svg>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="details-container limit"
    >
      <div
        class="loader flex"
      >
        <div
          aria-busy="true"
          aria-label="three-circles-rotating"
          class=""
          data-testid="three-circles-wrapper"
          role="status"
          style="display: flex;"
        >
          <svg
            data-testid="three-circles-svg"
            enable-background="new 0 0 100 100"
            height="100"
            version="1.1"
            viewBox="0 0 100 100"
            width="100"
            xml:space="preserve"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M31.6,3.5C5.9,13.6-6.6,42.7,3.5,68.4c10.1,25.7,39.2,38.3,64.9,28.1l-3.1-7.9c-21.3,8.4-45.4-2-53.8-23.3
  c-8.4-21.3,2-45.4,23.3-53.8L31.6,3.5z"
              fill="white"
            >
              <animatetransform
                attributeName="transform"
                attributeType="XML"
                dur="2s"
                from="0 50 50"
                repeatCount="indefinite"
                to="360 50 50"
                type="rotate"
              />
            </path>
            <path
              d="M42.3,39.6c5.7-4.3,13.9-3.1,18.1,2.7c4.3,5.7,3.1,13.9-2.7,18.1l4.1,5.5c8.8-6.5,10.6-19,4.1-27.7
  c-6.5-8.8-19-10.6-27.7-4.1L42.3,39.6z"
              fill="white"
            >
              <animatetransform
                attributeName="transform"
                attributeType="XML"
                dur="1s"
                from="0 50 50"
                repeatCount="indefinite"
                to="-360 50 50"
                type="rotate"
              />
            </path>
            <path
              d="M82,35.7C74.1,18,53.4,10.1,35.7,18S10.1,46.6,18,64.3l7.6-3.4c-6-13.5,0-29.3,13.5-35.3s29.3,0,35.3,13.5
  L82,35.7z"
              fill="white"
            >
              <animatetransform
                attributeName="transform"
                attributeType="XML"
                dur="2s"
                from="0 50 50"
                repeatCount="indefinite"
                to="360 50 50"
                type="rotate"
              />
            </path>
          </svg>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
